internal/settings.go

```diff
@@ -187,3 +188,32 @@ func (ds *DialSettings) GetUniverseDomain() string {
 func (ds *DialSettings) IsUniverseDomainGDU() bool {
 	return ds.GetUniverseDomain() == ds.GetDefaultUniverseDomain()
 }
+
+// GetUniverseDomain returns the default service domain for a given Cloud
+// universe, from google.Credentials, for comparison with the value returned by
+// (*DialSettings).GetUniverseDomain. This wrapper function should be removed
+// to close [TODO(chrisdsmith): issue link here]. See details below.
+func GetUniverseDomain(creds *google.Credentials) (string, error) {
+	timer := time.NewTimer(time.Second)
```

Comment:
Is one second long enough? Seems short, but I don't know the slo of the metadata service.

Reply:
@viacheslav-rostovtsev says 1s is appropriate.